solutions/advanced/cf-891C

```diff
@@ -0,0 +1,150 @@
+---
+id: cf-891C
+source: CF
+title: Envy
+author: Chongtian Ma
+---
+
+[Official Analysis](https://codeforces.com/blog/entry/55841)
+
+## Explanation
+
+This problem requires some MST intuition. In Kruskal's algorithm, we are to sort the edges by non-decreasing weight, and try uniting vertices if they aren't already in the same component. Here, we can try to check every query using this algorithm, by processing each edge of each query offline.
```

Comment:
what does the first sentence contribute

Reply:
that intuition is required to understand the following explanation

Reply:
this problem is in advanced though
it's like saying "you need prefix sum intuition to solve this" in a gold solution

Reply:
should i put this as a hint instead?

Reply:
idt that would even qualify as a hint tbh